plugins/documents/app/controllers/documents_controller.rb

```diff
@@ -0,0 +1,63 @@
+require 'filemagic'
+
+class DocumentsController < ApplicationController
+  before_filter -> { require_plugin_enabled FoodsoftDocuments }
+
+  def index
+    if params["sort"]
+      sort = case params["sort"]
+               when "name" then "name"
+               when "created_at" then "created_at"
+               when "name_reverse" then "name DESC"
+               when "created_at_reverse" then "created_at DESC"
+             end
+    else
+      sort = "name"
+    end
+
+    @documents = Document.page(params[:page]).per(@per_page).order(sort)
+  end
+
+  def new
+    @document = Document.new
+  end
+
+  def create
+    @document = Document.new
+    @document.data = params[:document][:data].read
+    @document.mime = FileMagic.new(FileMagic::MAGIC_MIME).buffer(@document.data)
```

Comment:
What do you think of including mimetype whitelisting here as well? This one can be a lot broader, but I'd be careful with executable files, for example. (Think of someone uploading an html file with javascript.)